doc/conf.py

```diff
@@ -403,6 +416,72 @@ def __call__(self, filename):
         return -version_float
 
 
+def notebook_modification_function(notebook_content, notebook_filename):
+    notebook_content_str = str(notebook_content)
+    warning_template = "\n".join(
+        [
+            "<div class='alert alert-{message_class}'>",
+            "",
+            "# JupyterLite warning",
+            "",
+            "{message}",
+            "</div>",
+        ]
+    )
+
+    message_class = "warning"
+    message = (
+        "Running the scikit-learn examples in JupyterLite is experimental and you may"
+        " encounter some unexpected behavior.\n\nThe main difference is that imports"
+        " will take a lot longer than usual, for example the first `import sklearn` can"
+        " take roughly 10-20s.\n\nIf you notice problems, feel free to open an"
+        " [issue](https://github.com/scikit-learn/scikit-learn/issues/new/choose)"
+        " about it."
+    )
+
+    markdown = warning_template.format(message_class=message_class, message=message)
+
+    dummy_notebook_content = {"cells": []}
+    add_markdown_cell(dummy_notebook_content, markdown)
+
+    code_lines = []
+
+    if "seaborn" in notebook_content_str:
+        code_lines.append("%pip install seaborn")
+    if "plotly.express" in notebook_content_str:
+        code_lines.append("%pip install plotly")
+    if "skimage" in notebook_content_str:
+        code_lines.append("%pip install scikit-image")
+    if "fetch_" in notebook_content_str:
+        code_lines.extend(
+            [
+                "%pip install pyodide-http",
+                "import pyodide_http",
+                "pyodide_http.patch_all()",
+            ]
+        )
+    # always import matplotlib and pandas to avoid Pyodide limitation with
+    # imports inside functions
+    code_lines.extend(["import matplotlib", "import pandas"])
+    # TODO temporary work-around for OpenML CORS headers issue
+    # https://github.com/openml/OpenML/issues/1135
+    code_lines.extend(
+        [
+            "from sklearn.datasets import _openml",
+            "_openml._OPENML_PREFIX = 'https://api.openml.org/'",
```

Comment:
With https://github.com/scikit-learn/scikit-learn/pull/26171 merged, is this patch required?

Reply:
Yes for now until the OpenML header thing is fixed.

Note that the scikit-learn version we are using is the version available in Pyodide, namely 1.2.2 at the time of writing and #26171 only is in `main`.

I have some medium-term plan to add a CI build for the scikit-learn development version, which potentially could be published similarly to our scikit-learn development wheel and installed in JupyterLite but it will take a bit of time.

Reply:
To check my understanding: if there was a Pyodide build that includes https://github.com/scikit-learn/scikit-learn/pull/26171, can this OpenML work-around be removed without waiting for https://github.com/openml/OpenML/issues/1135 to be resolved?

Reply:
Yes, hopefully the OpenML CORS issue can be fixed in the not too distant future and this temporary work-around can be remove soon.